src/plugins/hetero/executable_network.cpp

```diff
@@ -592,9 +592,10 @@ HeteroExecutableNetwork::HeteroExecutableNetwork(std::istream& heteroModel,
         }
 
         std::shared_ptr<ov::Node> node = std::make_shared<ov::op::v0::Parameter>(elementType, partialShape);
+        // For result operation_name is name of previous operation
+        node->set_friendly_name(operation_name);
         if (!is_param)
             node = std::make_shared<ov::op::v0::Result>(node);
-        node->set_friendly_name(operation_name);
```

Comment:
do we have a test for this change?

Reply:
It is reproduced by enabled test `smoke_Hetero_CachingSupportCase`